Libs/Widgets/ctkFileDialog.cpp

```diff
@@ -127,13 +131,10 @@ ctkFileDialog::ctkFileDialog(QWidget *parentWidget,
 {
   Q_D(ctkFileDialog);
 
-// The findChild<QDialogButtonBox*>() call fails on Mac/Qt5 because native
-// dialogs don't publish any internals. No problems on other OS.
-// Can be applied to Qt4 as well, if problems arise there.
-#if QT_VERSION >= QT_VERSION_CHECK(5,0,0)
-  this->setOptions(DontUseNativeDialog);
-#endif
-
+  if (!(this->options() & QFileDialog::DontUseNativeDialog))
+  {
+    return;  // Native dialog does not supporting modifying or getting widget elements which are done as part of init()
+  }
```

Comment:
I would have moved that condition in `init()` directly, in case there are (or will be added) other important operations in `init()` that are not related to to native/not native.